tracing-subscriber/src/registry/sharded.rs

```diff
@@ -0,0 +1,310 @@
+use sharded_slab::{Guard, Slab};
+
+use crate::{
+    fmt::span::SpanStack,
+    registry::{LookupSpan, SpanData},
+};
+use serde_json::{json, Value};
+use std::{
+    cell::RefCell,
+    collections::HashMap,
+    convert::TryInto,
+    fmt,
+    sync::{
+        atomic::{fence, AtomicUsize, Ordering},
+        Arc, Mutex,
+    },
+};
+use tracing_core::{
+    field::{FieldSet, Visit},
+    span::{self, Id},
+    Event, Field, Interest, Metadata, Subscriber,
+};
+
+#[derive(Debug)]
+pub struct Registry {
+    spans: Arc<Slab<Data>>,
+}
+
+thread_local! {
+    static CONTEXT: RefCell<SpanStack> = RefCell::new(SpanStack::new());
+}
+
+#[derive(Debug)]
+pub struct Data {
+    metadata: &'static Metadata<'static>,
+    parent: Option<Id>,
+    children: Vec<Id>,
+    ref_count: AtomicUsize,
+    // TODO(david): get rid of these
+    values: Mutex<HashMap<&'static str, Value>>,
+}
+
+impl Data {
+    fn drop_ref(&self, ordering: Ordering) -> bool {
```

Comment:
TIOLI: I'm on the fence about whether there should be a separate method on `Data` for this (as opposed to just moving it into `Registry::drop_ref`)

Reply:
Also, there's really no reason to pass in an `Ordering` here, it will always be `Release`...I think if we're going to have a separate method for this, having the method provide the ordering is probably a better separation of concerns.

Reply:
I put the drop_ref into `Registry`; removing additional & unnecessary `impl Data` block.

I also removed the `Ordering` param.

Reply:
@davidbarsky did you commit this? i don't see it?